backend/src/main/scala/cromwell/backend/standard/callcaching/RootWorkflowFileHashCacheActor.scala

```diff
@@ -0,0 +1,111 @@
+package cromwell.backend.standard.callcaching
+
+import java.io.IOException
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import cats.data.NonEmptyList
+import com.google.common.cache.{CacheBuilder, CacheLoader, LoadingCache}
+import cromwell.backend.standard.callcaching.RootWorkflowFileHashCacheActor.IoHashCommandWithContext
+import cromwell.core.actor.RobustClientHelper.RequestTimeout
+import cromwell.core.io._
+
+
+class RootWorkflowFileHashCacheActor private(override val ioActor: ActorRef) extends Actor with ActorLogging with IoClientHelper {
+  case class FileHashRequester(replyTo: ActorRef, fileHashContext: FileHashContext, ioCommand: IoCommand[_])
+
+  sealed trait FileHashValue
+  // The hash value is not yet in the cache and has not been requested.
+  case object FileHashValueNotRequested extends FileHashValue
+  // The hash value has been requested but is not yet in the cache.
+  case class FileHashValueRequested(requesters: NonEmptyList[FileHashRequester]) extends FileHashValue
+  // Hashing succeeded.
+  case class FileHashSuccess(value: String) extends FileHashValue
+  // Hashing failed.
+  case class FileHashFailure(error: String) extends FileHashValue
+
+  protected def ioCommandBuilder: IoCommandBuilder = DefaultIoCommandBuilder
```

Comment:
I think you would want any backend that has `gcsFileSystem` as a dependency (especially PAPI backends) to extend this class and override this with the `GcsBatchCommandBuilder`

Reply:
Actually this is vestigial cruft from an earlier factoring. The command building is now being done in the `StandardFileHashingActor` so I'll just delete this.